### PR TITLE
compiletest: Allow using revisions with debuginfo tests.

### PR DESCRIPTION
**Pull request by luqmana**
A small wart that came up in https://github.com/rust-lang/rust/pull/95685#issuecomment-1089184951.

### CODE REVIEW

File: src/tools/compiletest/src/header.rs [hunk 1]
```diff
@@ -535,6 +535,30 @@ impl TestProps {
...
+    comment: &str,
+    ln: &'line str,
+) -> Option<(Option<&'line str>, &'line str)> {
+    if ln.starts_with(comment) {
+        let ln = ln[comment.len()..].trim_start();
+        if ln.starts_with('[') {
+            // A comment like `//[foo]` is specific to revision `foo`
+            if let Some(close_brace) = ln.find(']') {
+                let open_brace = ln.find('[').unwrap();
+                let lncfg = &ln[open_brace + 1..close_brace];
```
Comment (tmiasko) [inline src/tools/compiletest/src/header.rs:548]: At this point `ln.starts_with('[')`, so `open_brace` is zero.

Comment (luqmana) [reply src/tools/compiletest/src/header.rs]: that is true 😅. thanks!

**Comment by tmiasko** [comment]
Could you squash "compiletest: Update line counter ..." into the earlier commit "compiletest: Allow using revisions ..." that introduced `continue`, and similarly squash "simplify" commit into the earlier commit that introduced the changed code.

Looks good to me. Thanks!

@bors delegate+

**Pull request merged**